content/concepts/introduction/protocols/overview.md

```diff
@@ -1,206 +1,109 @@
-
-The first takes two arguments: a protocol id, and a handler function. If an incoming stream request sends an exact
-match for the protocol id, the handler function will be invoked with the new stream as an argument.
-
-#### Using a match function
-
-The second kind of protocol registration takes three arguments: the protocol id, a protocol match function, and the handler function.
-
-When a stream request comes in whose protocol id doesn't have any exact matches, the protocol id will be passed through
-all of the registered match functions. If any returns `true`, the associated handler function will be invoked.
-
-This gives you a lot of flexibility to do your own "fuzzy matching" and define whatever rules for protocol matching
-make sense for your application.
-
-### Dialing a specific protocol
-
-When dialing a remote peer to open a new stream, the initiating peer sends the protocol id that they'd like to use. The remote peer will use
-the matching logic described above to accept or reject the protocol. If the protocol is rejected, the dialing peer can try again.
-
-When dialing, you can optionally provide a list of protocol ids instead of a single id. When you provide multiple protocol ids, they will
-each be tried in succession, and the first successful match will be used if at least one of the protocols is supported by the remote peer.
-This can be useful if you support a range of protocol versions, since you can propose the most recent version and fallback to older versions
-if the remote hasn't adopted the latest version yet.
+A new stream passes through a protocol multiplexer called
+[Multistream-select](multistream.md), which routes the stream to the appropriate
+protocol handler based on the protocol ID.
+
+### Handler functions
+
+A handler function handles each libp2p protocol, responsible
+for defining the protocol's behavior once the stream has been established.
+The handler function is invoked when an incoming stream is received with a
+registered protocol ID.
+
+The handler function can also specify a match function,
+which allows for the acceptance of non-exact string matches for protocol IDs.
+
+A libp2p application will define a stream handler that takes over the
+stream after protocol negotiation. Everything is sent and received after the
+application protocol defines the negotiation phase.
+
+### Binary streams
+
+The medium over which a protocol operates is a bi-directional
+binary stream. This stream provides the following features:
+
+- **Bidirectional, reliable delivery of binary data**: Both peers can read and write
+  from the stream at any time, and data is read in the same order as it was written.
+  The stream can also be "half-closed", meaning it can be closed for writing while
+  still open for reading or closed for reading while still open for writing.
+- **Supports backpressure**: Eager writers cannot flood readers with data, as the
+  stream automatically regulates data flow to prevent overload.
```

Comment:
```suggestion
- **Supports backpressure**: Eager writers cannot flood readers with data, as the
  stream automatically regulates data flow to prevent overload.
```
Not by default, if TCP + mplex is used, there's no support for backpressure. I think this existed in the legacy document but it should be deleted